### PR TITLE
Ported to Python 3

### PR DESCRIPTION
**Pull request by frafra**
`make` and `python3 testpyisomd5sum.py` seem to work correctly.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
-PYVER  := $(shell python -c 'import sys; print sys.version[0:3]')
+PYVER  := $(shell python3 -c 'import sys; print(sys.version[0:3])')
 PYTHON = python$(PYVER)
-PYTHONINCLUDE = /usr/include/$(PYTHON)
+PYTHONINCLUDE = /usr/include/$(PYTHON)m
```
Comment (vpodzime) [inline Makefile]: Is this change intentional? If so, what's the purpose of it?

File: pyisomd5sum.c [hunk 2]
```diff
@@ -86,7 +86,14 @@ static PyObject * doImplantIsoMD5Sum(PyObject * s, PyObject * args) {
     return Py_BuildValue("i", rc);
 }
 
-
-void initpyisomd5sum(void) {
-    (void)Py_InitModule("pyisomd5sum", isomd5sumMethods);
+static struct PyModuleDef pyisomd5sum = {
+    PyModuleDef_HEAD_INIT,
+    "pyisomd5sum", /* name of module */
+    "",          /* module documentation, may be NULL */
```
Comment (vpodzime) [inline pyisomd5sum.c]: Why not ``NULL`` then?

**Comment by vpodzime** [comment]
Looks good to me other than the above questions.

**Comment by frafra** [comment]
"m" was intentional (I saw 3.4m on my Fedora machines), but I replaced it with something more portable and robust. 

I also used NULL instead of an empty string.

**Comment by vpodzime** [comment]
Looks great to me now. Thanks!

**Comment by frafra** [comment]
Thank you :)
Here's a minor patch for the Makefile, in order to make it more robust.

**Comment by vpodzime** [comment]
Looks good to me too.

**Comment by bcl** [comment]
Do we want to make this py2 and py3 compatible? Either way, the version should be bumped to 2.0.0

**Comment by frafra** [comment]
I would say that is not strictly necessary, because isomd5sum is not a library.

**Comment by rfinnie** [comment]
Supporting both 2 and 3 would make it easier for Debian downstream, as the Python module is in a separate binary package (python-pyisomd5sum).  Debian Python packages use python-* for Python 2 and python3-* for Python 3, so a package rename + dropping support for the Python 2 version would be required.

See SHA: rfinnie@8c4fe4188eae16f44fb120277aa25340d0699ada which reintroduces $(PYTHON) (defaults to python3), and pyisomd5sum.c checks for PYTHON_ABI_VERSION (which was introduced with Python 3) in the preprocessor, allowing for either to work:

```
(sid)ryan@chroot-dev:~/git/isomd5sum{master}$ make pyisomd5sum.o clean
cc  -Wall -Werror -D_GNU_SOURCE=1 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE=1 -D_LARGEFILE64_SOURCE=1 -fPIC -I/usr/include/python3.4m -I/usr/include/python3.4m -c -O -o pyisomd5sum.o pyisomd5sum.c
rm -f *.o *.so *.pyc *.a .depend *~
rm -f implantisomd5 checkisomd5 
(sid)ryan@chroot-dev:~/git/isomd5sum{master}$ make pyisomd5sum.o clean PYTHON=python2.7
cc  -Wall -Werror -D_GNU_SOURCE=1 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE=1 -D_LARGEFILE64_SOURCE=1 -fPIC -I/usr/include/python2.7 -I/usr/include/x86_64-linux-gnu/python2.7 -c -O -o pyisomd5sum.o pyisomd5sum.c
rm -f *.o *.so *.pyc *.a .depend *~
rm -f implantisomd5 checkisomd5 
```

**Comment by bcl** [comment]
Thanks to everyone for the patches. They have been merged upstream and a new fedora package built for rawhide. I did make one more change to the Makefile, PYTHON ?= python3 allows it to be overridden by the environment.
